src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -300,8 +319,92 @@ public void visitParameter(JdbcParameter jdbcParameter) {
     }
 
     @Override
+    @SuppressWarnings("OperatorPrecedence")
     public void visitSelectStatement(SelectStatement selectStatement) {
-        throw new FeatureNotSupportedException("TODO-HIBERNATE-22 https://jira.mongodb.org/browse/HIBERNATE-22");
+        logSqlAst(selectStatement);
+        if (!selectStatement.getQueryPart().isRoot()) {
+            throw new FeatureNotSupportedException("Subquery not supported");
+        }
+        if (selectStatement.getCteStatements() != null
+                        && !selectStatement.getCteStatements().isEmpty()
+                || selectStatement.getCteObjects() != null
+                        && !selectStatement.getCteObjects().isEmpty()) {
+            throw new FeatureNotSupportedException("CTE feature not supported");
+        }
+        selectStatement.getQueryPart().accept(this);
```

Comment:
QueryPart has two child classes:

- QueryGroup
- QuerySpec

The latter will be covered in the visitor method below; QueryGroup is used for a set of QuerySpec and how to combine them together. A typical example is as follows:

```
SELECT column_name(s) FROM table1
UNION
SELECT column_name(s) FROM table2;
```